### PR TITLE
#18990 now the key value on the url map as a map instead of single st…

### PR DESCRIPTION
**Pull request by jdotcms**
Now on the page render api; the key value on the url map is being rendered as a map (json map) instead of a string

### CODE REVIEW

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.anyString;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:21]: Codacy found an issue: [Avoid unused imports such as 'org.mockito.Matchers.anyString'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    /**
+     * Method to test: createObjectMapKeyValue
+     * Given Scenario: key value but not key field
+     * ExpectedResult: return the key value as string
+     */
+    @Test
+    public void createObjectMapUrlContent_convert_url_map_no_keyvalue() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:44]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
+package com.dotmarketing.portlets.htmlpageasset.business.render.page;
+
+import com.dotcms.contenttype.model.field.KeyValueField;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.workflows.model.WorkflowScheme;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:9]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.workflows.model.WorkflowScheme'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+     */
+    @Test
+    public void createObjectMapUrlContent_convert_url_map_no_keyvalue() throws Exception {
+
+        final PageViewSerializer pageViewSerializer = new PageViewSerializer();
+        final Map<String, Object> pageViewMap       = new HashMap<>();
+        final Contentlet urlContent                 = new Contentlet();
+        final String     keyValue                   = "{\"key1\":\"value1\", \"key2\":\"value2\"}";
+        urlContent.getMap().put("title", "Test");
+        urlContent.getMap().put("keyValue", keyValue);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:51]: Codacy found an issue: [The String literal "keyValue" appears 6 times in this file; the first occurrence is on line 51](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
+package com.dotmarketing.portlets.htmlpageasset.business.render.page;
+
+import com.dotcms.contenttype.model.field.KeyValueField;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.business.APILocator;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:6]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.business.APILocator'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+import com.dotmarketing.business.PermissionAPI;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:7]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.business.PermissionAPI'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+import static org.mockito.Mockito.when;
+
+/**
+ * Test for the {@link PageViewSerializerTest}
+ * @author jsanca
+ */
+public class PageViewSerializerTest {
+
+    @BeforeClass
+    public static void prepare () throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:32]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        assertTrue(pageViewMap.isEmpty());
+    }
+
+    /**
+     * Method to test: createObjectMapKeyValue
+     * Given Scenario: key value given
+     * ExpectedResult: return the key value as map
+     */
+    @Test
+    public void createObjectMapUrlContent_convert_url_map_keyvalue() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:66]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java [hunk 9]
```diff
@@ -54,14 +60,42 @@ public void serialize(final PageView pageView, final JsonGenerator jsonGenerator
...
+    }
+
+    /**
+     * Get all key value fields into the contentlet (if any) and put as an object into the contentMap (returns it)
+     * @param contentlet    {@link Contentlet}
+     * @param contentType   {@link ContentType}
+     * @param contentletMap {@link Map}
+     * @return Map
+     */
+    protected Map<String, Object> createObjectMapKeyValue(final Contentlet contentlet, final ContentType contentType, final Map<String, Object> contentletMap) {
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java:85]: I think this code should be part of a more generic Serializer - `DotContentletTransformer`- any content can have a key/value field, not just Pages

**Review by wezell — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java [hunk 10]
```diff
@@ -1,9 +1,15 @@
...
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.contenttype.model.field.KeyValueField;
+import com.dotcms.contenttype.model.type.ContentType;
 import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
 import com.dotmarketing.portlets.contentlet.transform.DotContentletTransformer;
 import com.dotmarketing.portlets.contentlet.transform.DotTransformerBuilder;
+import com.dotmarketing.portlets.contentlet.transform.strategy.KeyValueViewStrategy;
 import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.UtilMethods;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java:12]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.util.UtilMethods'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+import static org.mockito.Mockito.when;
+
+/**
+ * Test for the {@link PageViewSerializerTest}
+ * @author jsanca
+ */
+public class PageViewSerializerTest {
+
+    @BeforeClass
+    public static void prepare () throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:28]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java [hunk 12]
```diff
@@ -13,10 +19,13 @@
 import java.io.CharArrayReader;
 import java.io.IOException;
 import java.util.Collections;
+import java.util.List;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java:22]: Codacy found an issue: [Avoid unused imports such as 'java.util.List'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+        assertTrue(pageViewMap.isEmpty());
+    }
+
+    /**
+     * Method to test: createObjectMapKeyValue
+     * Given Scenario: key value given
+     * ExpectedResult: return the key value as map
+     */
+    @Test
+    public void createObjectMapUrlContent_convert_url_map_keyvalue() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:63]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java [hunk 14]
```diff
@@ -1,9 +1,15 @@
 package com.dotmarketing.portlets.htmlpageasset.business.render.page;
 
+import com.dotcms.contenttype.model.field.Field;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java:3]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.contenttype.model.field.Field'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java [hunk 15]
```diff
@@ -1,9 +1,15 @@
...
+import com.dotcms.contenttype.model.field.KeyValueField;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java:4]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.contenttype.model.field.KeyValueField'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java [hunk 16]
```diff
@@ -13,10 +19,13 @@
 import java.io.CharArrayReader;
 import java.io.IOException;
 import java.util.Collections;
+import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.TreeMap;
 
+import static java.util.Collections.emptyMap;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java:27]: Codacy found an issue: [Avoid unused imports such as 'java.util.Collections.emptyMap'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 17]
```diff
@@ -0,0 +1,86 @@
...
+     */
+    @Test
+    public void createObjectMapUrlContent_convert_url_map_no_keyvalue() throws Exception {
+
+        final PageViewSerializer pageViewSerializer = new PageViewSerializer();
+        final Map<String, Object> pageViewMap       = new HashMap<>();
+        final Contentlet baseContent                = new Contentlet();
+        final Contentlet urlContent                 = mock(Contentlet.class);
+        final String     keyValue                   = "{\"key1\":\"value1\", \"key2\":\"value2\"}";
+        baseContent.getMap().put("keyValue", keyValue);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:47]: Codacy found an issue: [The String literal "keyValue" appears 10 times in this file; the first occurrence is on line 47](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 18]
```diff
@@ -0,0 +1,86 @@
...
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    /**
+     * Method to test: createObjectMapKeyValue
+     * Given Scenario: key value but not key field
+     * ExpectedResult: return the key value as string
+     */
+    @Test
+    public void createObjectMapUrlContent_convert_url_map_no_keyvalue() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:40]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java [hunk 19]
```diff
@@ -54,14 +63,36 @@ public void serialize(final PageView pageView, final JsonGenerator jsonGenerator
...
+     * @param contentType   {@link ContentType}
+     * @param contentletMap {@link Map}
+     * @return Map
+     */
+    protected Map<String, Object> createObjectMapKeyValue(final Contentlet contentlet,
+                                                          final Map<String, Object> contentletMap) {
+
+        new KeyValueViewStrategy(null).apply(contentlet, contentletMap, null, null);
+        return contentletMap;
+    }
```
Comment (fabrizzio-dotCMS) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java:93]: I don't think the strategy should be applied directly it must be applied from within the transformer. 

Let's say you do
`new DotTransformerBuilder().defaultOptions() `
the code that instructs the transformer to apply this strategy must be part of the options provided to the transformer. You should never apply a strategy outside of the transformer

**Review by fabrizzio-dotCMS — CHANGES_REQUESTED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 20]
```diff
@@ -0,0 +1,98 @@
...
+        final ContentType contentType               = mock(ContentType.class);
+        when(contentType.baseType()).thenReturn(BaseContentType.CONTENT);
+        when(urlContent.getTitleImage()).thenReturn(Optional.empty());
+        when(urlContent.getContentType()).thenReturn(contentType);
+        when(urlContent.getBaseType()).thenReturn(Optional.of(BaseContentType.CONTENT));
+        when(urlContent.get("keyValue")).thenReturn(keyValue);
+        when(contentType.fields(KeyValueField.class)).thenReturn(new ArrayList<>());
+        pageViewSerializer.createObjectMapUrlContent(urlContent, pageViewMap);
+
+        assertTrue(pageViewMap.containsKey("urlContentMap"));
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:61]: Codacy found an issue: [The String literal "urlContentMap" appears 6 times in this file; the first occurrence is on line 61](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java [hunk 21]
```diff
@@ -1,9 +1,15 @@
 package com.dotmarketing.portlets.htmlpageasset.business.render.page;
 
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.contenttype.model.field.KeyValueField;
+import com.dotcms.contenttype.model.type.ContentType;
 import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
 import com.dotmarketing.portlets.contentlet.transform.DotContentletTransformer;
 import com.dotmarketing.portlets.contentlet.transform.DotTransformerBuilder;
+import com.dotmarketing.portlets.contentlet.transform.strategy.KeyValueViewStrategy;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java:10]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.contentlet.transform.strategy.KeyValueViewStrategy'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java [hunk 22]
```diff
@@ -1,9 +1,15 @@
 package com.dotmarketing.portlets.htmlpageasset.business.render.page;
 
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.contenttype.model.field.KeyValueField;
+import com.dotcms.contenttype.model.type.ContentType;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java:5]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.contenttype.model.type.ContentType'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 23]
```diff
@@ -0,0 +1,98 @@
...
+     */
+    @Test
+    public void createObjectMapUrlContent_convert_url_map_no_keyvalue() throws Exception {
+
+        final PageViewSerializer pageViewSerializer = new PageViewSerializer();
+        final Map<String, Object> pageViewMap       = new HashMap<>();
+        final Contentlet baseContent                = new Contentlet();
+        final Contentlet urlContent                 = mock(Contentlet.class);
+        final String     keyValue                   = "{\"key1\":\"value1\", \"key2\":\"value2\"}";
+        baseContent.getMap().put("keyValue", keyValue);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:51]: Codacy found an issue: [The String literal "keyValue" appears 10 times in this file; the first occurrence is on line 47](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java:12]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.util.UtilMethods'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 24]
```diff
@@ -0,0 +1,98 @@
...
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    /**
+     * Method to test: createObjectMapKeyValue
+     * Given Scenario: key value but not key field
+     * ExpectedResult: return the key value as string
+     */
+    @Test
+    public void createObjectMapUrlContent_convert_url_map_no_keyvalue() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:44]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 25]
```diff
@@ -0,0 +1,98 @@
...
+        when(urlContent.get("keyValue")).thenReturn(keyValue);
+        when(contentType.fields(KeyValueField.class)).thenReturn(Arrays.asList(keyValueField));
+        when(keyValueField.variable()).thenReturn("keyValue");
+        when(urlContent.getKeyValueProperty("keyValue")).thenReturn(baseContent.getKeyValueProperty("keyValue"));
+        pageViewSerializer.createObjectMapUrlContent(urlContent, pageViewMap);
+
+        assertFalse(pageViewMap.isEmpty());
+        assertEquals(1, pageViewMap.size());
+        assertTrue(pageViewMap.containsKey("urlContentMap"));
+        assertTrue((Map.class.cast(pageViewMap.get("urlContentMap")).containsKey("keyValue")));
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:94]: Codacy found an issue: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 26]
```diff
@@ -0,0 +1,98 @@
...
+import static org.mockito.Mockito.when;
+
+/**
+ * Test for the {@link PageViewSerializerTest}
+ * @author jsanca
+ */
+public class PageViewSerializerTest {
+
+    @BeforeClass
+    public static void prepare () throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:32]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 27]
```diff
@@ -0,0 +1,98 @@
+package com.dotmarketing.portlets.htmlpageasset.business.render.page;
+
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.contenttype.model.field.KeyValueField;
+import com.dotcms.contenttype.model.field.TextField;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:5]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.contenttype.model.field.TextField'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java (hunk 12 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java:22]: Codacy found an issue: [Avoid unused imports such as 'java.util.List'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java (hunk 15 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java:4]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.contenttype.model.field.KeyValueField'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java (hunk 14 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java:3]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.contenttype.model.field.Field'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 28]
```diff
@@ -0,0 +1,98 @@
...
+        assertFalse(Map.class.cast(pageViewMap.get("urlContentMap")).containsKey("keyValue"));
+    }
+
+    /**
+     * Method to test: createObjectMapKeyValue
+     * Given Scenario: key value given
+     * ExpectedResult: return the key value as map
+     */
+    @Test
+    public void createObjectMapUrlContent_convert_url_map_keyvalue() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:71]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java (hunk 16 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java:27]: Codacy found an issue: [Avoid unused imports such as 'java.util.Collections.emptyMap'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java [hunk 29]
```diff
@@ -0,0 +1,98 @@
+package com.dotmarketing.portlets.htmlpageasset.business.render.page;
+
+import com.dotcms.contenttype.model.field.Field;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializerTest.java:3]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.contenttype.model.field.Field'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6197346)

**Pull request merged**